api/opencl_platform_layer.asciidoc

```diff
@@ -1167,18 +1167,20 @@ include::{generated}/api/version-notes/CL_DEVICE_EXTENSIONS.asciidoc[]
         *cl_khr_local_int32_extended_atomics*
 
         Additionally, the following Khronos extension names must be returned
-        by all devices that support OpenCL 1.2:
+        by all devices that support OpenCL 1.2 when the optional feature is
+        supported:
 
         *cl_khr_fp64* (if double precision is supported)
 
         Additionally, the following Khronos extension names must be returned
         by all devices that support OpenCL 2.0, OpenCL 2.1, or OpenCL 2.2.
-        Note that these Khronos extension names are *not* required for
-        devices supporting OpenCL 3.0:
+        For devices that support OpenCL 3.0, these extension names must only
+        be returned when the optional feature is supported:
 
-        *cl_khr_3d_image_writes* +
-        *cl_khr_depth_images* +
-        *cl_khr_image2d_from_buffer*
+        *cl_khr_3d_image_writes* (if 3D image writes are supported) +
+        *cl_khr_depth_images* (if depth images are supported) +
```

Comment:
Do we want to define what support for depth images mean? I note there's a bit of a contrast between what we say in Appendix H for 3d image writes and depth images :). Neither of which are normative BTW.

Reply:
I updated Appendix H so it is consistent for 3D Image Writes and Depth Images (and sRGB images) now.

I didn't precisely define what is meant by "support for depth images" though.  If we to define this I'm open to it, but I don't think it's strictly necessary, and I don't think it belongs here in the description for `CL_DEVICE_EXTENSIONS` anyhow.

Reply:
The change is definitely an upgrade but the next question becomes: what does it mean for the feature to be supported with no formats returned :).

Reply:
I know the appendix is non-normative, but we do cover this case there (e.g. "Will not return any image formats with `image_channel_order` equal to `CL_DEPTH` if no devices in _context_ support Depth Images."), and we'll have a CTS test that tests for strange combinations like reporting the extension but no image formats.  I hope we're OK :)

Reply:
@kpet Is this sufficient for you?